app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java

```diff
@@ -148,9 +162,105 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
+     * <p>
+     * This method is called after the {@link ManualUpdateDescriptorsWorker} is enqueued.
+     * The {@link ManualUpdateDescriptorsWorker} task is to fetch updates for the descriptors.
+     * <p>
+     * If the task is successful, the {@link WorkInfo} object will contain the updated descriptors.
+     * Otherwise, the {@link WorkInfo} object will be null.
+     *
+     * @param workInfo The {@link WorkInfo} of the task.
+     */
+    private void onManualUpdatesFetchComplete(WorkInfo workInfo) {
+        if (workInfo != null) {
+            binding.reviewUpdateNotificationFragment.setVisibility(View.VISIBLE);
+            switch (workInfo.getState()) {
+                case SUCCEEDED -> getSupportFragmentManager()
+                        .beginTransaction()
+                        .add(
+                                R.id.review_update_notification_fragment,
+                                OONIRunDynamicProgressBar.newInstance(ProgressType.REVIEW_LINK, new OnActionListener() {
+                                    @Override
+                                    public void onActionButtonCLicked() {
+
+                                        getReviewUpdatesLauncher().launch(
+                                                ReviewDescriptorUpdatesActivity.newIntent(
+                                                        MainActivity.this,
+                                                        workInfo.getOutputData().getString(ManualUpdateDescriptorsWorker.KEY_UPDATED_DESCRIPTORS)
+                                                )
+                                        );
+                                        removeProgressFragment(R.id.review_update_notification_fragment);
+                                    }
+
+                                    @Override
+                                    public void onCloseButtonClicked() {
+                                        removeProgressFragment(R.id.review_update_notification_fragment);
+                                    }
+                                }),
+                                OONIRunDynamicProgressBar.getTAG() + "_review_update_success_notification"
+                        ).commit();
+                case ENQUEUED -> getSupportFragmentManager()
+                        .beginTransaction()
+                        .add(
+                                R.id.review_update_notification_fragment,
+                                OONIRunDynamicProgressBar.newInstance(ProgressType.UPDATE_LINK, null),
+                                OONIRunDynamicProgressBar.getTAG() + "_review_update_enqueued_notification"
+                        ).commit();
+                case FAILED -> Snackbar.make(
+                        binding.getRoot(),
+                        R.string.Modal_Error,
+                        Snackbar.LENGTH_LONG
+                ).setAnchorView(binding.bottomNavigation).show();
+                default -> {
```

Comment:
```suggestion

                default -> {
```

I think we should document what is the default case with a comment. Is default === no updates?